src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -107,9 +107,20 @@ public async Task ConnectAsync(Uri uri, HttpMessageInvoker? invoker, Cancellatio
                             externalAndAbortCancellation = _abortSource;
                         }
 
+                        Task<HttpResponseMessage> sendTask;
+                        if (invoker is HttpClient)
+                        {
+                            HttpClient? client = invoker as HttpClient;
+                            sendTask = client!.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, externalAndAbortCancellation.Token);
+                        }
+                        else
+                        {
+                            sendTask = invoker.SendAsync(request, externalAndAbortCancellation.Token);
+                        }
+
                         using (linkedCancellation)
                         {
```

Comment:
```suggestion
                        using (linkedCancellation)
                        {
                            Task<HttpResponseMessage> sendTask = invoker is HttpClient client
                                ? client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, externalAndAbortCancellation.Token)
                                : invoker.SendAsync(request, externalAndAbortCancellation.Token);
```

The call should happen inside the `using` so that the cancellation is disposed even if it throws synchronously.